pkg/apis/core/validation/shoot.go

```diff
@@ -1211,6 +1212,8 @@ func ValidateWorker(worker core.Worker, kubernetesVersion string, fldPath *field
 
 		if worker.Kubernetes.Kubelet != nil {
 			allErrs = append(allErrs, ValidateKubeletConfig(*worker.Kubernetes.Kubelet, kubernetesVersion, isDockerConfigured([]core.Worker{worker}), fldPath.Child("kubernetes", "kubelet"))...)
+		} else if kubernetes.Kubelet != nil {
+			allErrs = append(allErrs, ValidateKubeletConfig(*kubernetes.Kubelet, kubernetesVersion, isDockerConfigured([]core.Worker{worker}), fldPath.Child("kubernetes", "kubelet"))...)
```

Comment:
Isn't this already done here? https://github.com/gardener/gardener/blob/583ce9a075f06228223ace8354e31576a8c82a3d/pkg/apis/core/validation/shoot.go#L841-L843

Why do we need to do it again?

Reply:
The difference in those two validations is that in the new one we are validating with the worker's kubernetes version which might be different from the control plane's one.
Without the new validation we can have a case where the control plane's version is 1.25 with set `seccompDefault` kubelet configuration to `true` and the workers version is 1.24 without any kubelet configurations. In this case the validation would not give any errors even though the worker will have invalid kubelet configuration when it inherits it form the control plane.